src/upgrade.py

```diff
@@ -77,6 +80,49 @@ def log_rollback_instructions(self) -> None:
             "Run `juju refresh --revision <previous-revision> postgresql` to initiate the rollback"
         )
 
+    def _on_upgrade_charm_check_legacy(self) -> None:
+        if not self.peer_relation or len(self.app_units) < len(self.charm.app_units):
+            logger.debug("Wait all units join the upgrade relation")
+            return
+
+        if self.state:
+            # If state set, upgrade is supported. Just set the snap information
+            # in the dependencies, as it's missing in the first revisions that
+            # support upgrades.
+            dependencies = self.peer_relation.data[self.charm.app].get("dependencies")
+            if (
+                self.charm.unit.is_leader()
+                and dependencies is not None
+                and "snap" not in json.loads(dependencies)
+            ):
+                fixed_dependencies = json.loads(dependencies)
+                fixed_dependencies["snap"] = {
+                    "dependencies": {},
+                    "name": "charmed-postgresql",
+                    "upgrade_supported": "^14",
+                    "version": "14.9",
```

Comment:
one more place to update on snap bump? :-(

Reply:
I don't think this should move, as far as I understand, it is for charm revs that don't have it defined.

Reply:
Exactly as Dragomir said. It's here only for the revisions missing the `snap` field in the dependency model.